### PR TITLE
Remove interfaces and channel usage

### PR DESCRIPTION
**Pull request by ldelossa**
This PR removes the interfaces present in libindex and libvuln along
with removing the channel usage at the API boundary.

Comments are cleanead up and the use of "Scan" is replaced with "Index"
where applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,25 +26,16 @@ m := &claircore.Manifest{
...
-srC, err := lib.Scan(m)
+ir, err := lib.Index(m)
 if err != nil {
     log.Printf("%v", err)
 }
-
-// block on channel
-sr := <-srC
-if sr.State == "ScannError" {
+if ir.State == "ScannError" {
```
Comment (hdonnay) [inline README.md]: two "n"s?

File: libvuln/libvuln.go [hunk 2]
```diff
@@ -13,13 +13,8 @@ import (
 	"github.com/quay/claircore/libvuln/driver"
 )
 
-// Libvuln is an interface exporting the public methods of our library.
-type Libvuln interface {
-	Scan(ctx context.Context, sr *claircore.IndexReport) (*claircore.VulnerabilityReport, error)
-}
-
 // libvuln implements the libvuln.Lubvuln interface
```
Comment (hdonnay) [inline libvuln/libvuln.go]: This interface doesn't exist anymore

File: README.md (hunk 1 above)
Comment (ldelossa) [reply README.md]: fixed

File: libvuln/libvuln.go (hunk 2 above)
Comment (ldelossa) [reply libvuln/libvuln.go]: fixed

**Pull request merged**